lib/Sema/CSDiag.cpp

```diff
@@ -7245,6 +7246,11 @@ visitRebindSelfInConstructorExpr(RebindSelfInConstructorExpr *E) {
   return false;
 }
 
+bool FailureDiagnosis::visitCaptureListExpr(CaptureListExpr *CLE) {
```

Comment:
@NachoSoto https://github.com/apple/swift/blob/master/include/swift/AST/ASTVisitor.h#L57 based on the kind of expression, it's going to get dispatched to the matching method.